### ISSUE TITLE
netcdf @rpath issue causes vtk problems

### ISSUE THREAD

**Issue by tdsmith** [Homebrew/homebrew-science#5173]
Importing vtk with system python fails. vtk masks the real error message but after removing the try/except in vtkIOExodus.py the error is:

```
tim@rocketman:~$ PYTHONPATH=/usr/local/lib/python2.7/site-packages /usr/bin/python -c 'import vtk'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/vtk/__init__.py", line 101, in <module>
    from .vtkIOExodus import *
  File "/usr/local/lib/python2.7/site-packages/vtk/vtkIOExodus.py", line 4, in <module>
    from .vtkIOExodusPython import *
ImportError: dlopen(/usr/local/lib/python2.7/site-packages/vtk/vtkIOExodusPython.so, 2): Library not loaded: @rpath/libnetcdf.11.dylib
  Referenced from: /usr/local/opt/netcdf/lib/libnetcdf_c++.4.dylib
  Reason: image not found
```

The linkage of libnetcdf_c++.dylib is:
```
tim@rocketman:lib$ otool -L libnetcdf_c++.4.dylib 
libnetcdf_c++.4.dylib:
	/usr/local/opt/netcdf/lib/libnetcdf_c++.4.dylib (compatibility version 7.0.0, current version 7.0.0)
	@rpath/libnetcdf.11.dylib (compatibility version 11.0.0, current version 11.4.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.4.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.0.0)
```

Using an absolute path instead of an @rpath with `install_name_tool -change @rpath/libnetcdf.11.dylib $PWD/libnetcdf.11.dylib libnetcdf_c++.4.dylib` allows the import to work.

I think this is a bad interaction between @rpath and System Integrity Protection.

Some options:
* close as wontfix because don't use system Python
* rewrite the @rpath to an absolute path at install time

**Comment by iMichka** [comment]
I think the rpath user got some mails ... :)

I am willing to help debug/test this. I do not really care about system Python, but netcdf is user at other places and could make trouble in other situations I guess.
@tdsmith do you have something in mind on how to rewrite that path? Is this something that needs to be done somewhere in the netcdf setup code?

**Comment by ilovezfs** [comment]
I've opened #5184.

**Comment by tdsmith** [comment]
Closed by 77fb5db.
